pymedphys/_dicom/delivery/core.py

```diff
@@ -12,21 +12,29 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import os
 import textwrap
 from copy import deepcopy
+from typing import Union, cast
 
 from pymedphys._imports import numpy as np
 from pymedphys._imports import pydicom
 
 from pymedphys._base.delivery import DeliveryBase
+from pymedphys._dicom import rtplan as _pmp_rtplan
+from pymedphys._dicom.delivery import utilities
 from pymedphys._utilities.transforms import convert_IEC_angle_to_bipolar
 
-from .. import rtplan
-from . import utilities
+dicom_path_or_dataset = Union[os.PathLike, "pydicom.Dataset"]
 
 
-def load_dicom_file(filepath):
+class NotSupportedError(ValueError):
```

Comment:
It's a nit, but "NotSupported" in other languages is used to denote that something hasn't been implemented yet, or in some edge cases, that a child class considers a particular method inappropriate even though the parent supports it.

So... this is ValueNotSupported(ValueError), and it is distinguished from ValueError based on what criteria? (which would be good to have in a docstring). e.g. "While the Value is valid in the sense of being part of a Defined or Enumerated type, it is not acceptable in the current circumstances"

Reply:
Yeah, Python has "NotImplementedError" which I was avoiding using. Here, the custom error class was so that I could explicitly catch this error and no other within the GUI. See where I have caught that error within streamlit:

https://github.com/pymedphys/pymedphys/blob/5f569a1be141cdd30cef94224c7e00b80568a743/pymedphys/_streamlit/apps/mudensity/_dicom.py#L159-L181

More than happy for a different name. In every sense of the word it is still a ValueError (which is why I am deriving from it), but I don't want a different ValueError to begin down the pathway designated within the GUI in that code linked in above.